### DISCUSSION TITLE
Device Unlocking without keystore (Different passcode to unlock databses, different passcodes for each database)

### DISCUSSION THREAD

**Discussion by trymeouteh** [Kunzisoft/KeePassDX#1738] [Ideas]
Please add a setting to not use the Android keystore and instead have a built in encrypted device unlock feature.

A few benefits of this will be...
- Using a different passcode for keepass databases than your device passcode!
- A unique passcode for each keepass database!
- Shorter password instead of a passcode (Your keepass database password could be 100 characters, but your unlock password in keepassdx can be 12 characters and be able to use letters!)

Also this will be useful if a users device does not have Android Keystore or have Android Keystore installed securely on their device.

**Comment by J-Jamet** [comment]
> Please add a setting to not use the Android keystore and instead have a built in encrypted device unlock feature

The Keystore is specifically designed to manage encrypted keys, so not using it would mean not encrypting the keys.
https://developer.android.com/privacy-and-security/keystore
